identity/src/main/java/com/android/identity/DeviceResponseParser.java

```diff
@@ -147,35 +148,30 @@ public static final class DeviceResponse {
         private @NonNull
         Pair<PublicKey, Map<String, Map<Long, byte[]>>> parseMso(DataItem mso,
                                                                  String expectedDoctype) {
+            byte[] encodedMSO = Util.cborEncode(mso);
+            MobileSecurityObjectParser.MobileSecurityObject parsedMso = new MobileSecurityObjectParser()
+                    .setMobileSecurityObject(encodedMSO).parse();
```

Comment:
Can we change `parseMso()` to take a `byte[] encodedMso` instead? It looks like the caller is currently decoding it just so we can reencode it... would be nice to avoid that.

Reply:
I see what you're saying, but the caller decodes so that they can call `Util.cborExtractTaggedAndEncodedCbor` before encoding (like in `MobileSecurityObjectParserTest` lines 45-49). I made it more consistent with this new commit by changing the input to `parseMso()` to a `byte[] encodedMso` and moved the encoding outside the function - what do you think?

Reply:
Latest code seems to parse to MSO multiple times, including both in `parseMso()` and `parseValidityInfo()` and this seems inefficient.  It's also a little weird to have a function called `parseMso()` because we now have a class that does all this. Maybe it's clearer to move both into the caller (`parseIssuerSigned()`)...

Reply:
Agree, I just wasn't sure how much to change the exiting structure, but I updated it with the newest commit